servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/DefaultBlockingIterableProcessor.java

```diff
@@ -127,7 +100,9 @@ public T next() {
             if (!hasNext()) {
                 throw new NoSuchElementException();
             }
-            return processNext();
+            T next = this.next;
+            this.next = null;
+            return next;
```

Comment:
these 3 lines of code are the same as the change of line 92... Consider sharing the code via a private method:

```java
private T consumeNext() {
    T next = this.next;
    this.next = null;
    return null;
}
```